app/Infrastructure/Exceptions/Handler.php

```diff
@@ -37,4 +41,21 @@ public function register()
             //
         });
     }
+
+    /**
+     * @param Request $request
+     * @param Throwable $userNotFoundExcepction
+     * @return Response
+     * @throws Throwable
+     */
+    public function render($request, Throwable $userNotFoundExcepction): Response
```

Comment:
Now you only have this exception and is correct, but when we scale this is not gonna be valid and we are gonna need a generic name.
```suggestion
    public function render($request, Throwable $userNotFoundException): Response
```